scrapy/pipelines/images.py

```diff
@@ -203,49 +188,25 @@ def get_images(
                 f"{self.min_width}x{self.min_height})"
             )
 
-        if self._deprecated_convert_image is None:
-            self._deprecated_convert_image = "response_body" not in get_func_args(
-                self.convert_image
-            )
-            if self._deprecated_convert_image:
-                warnings.warn(
-                    f"{self.__class__.__name__}.convert_image() method overridden in a deprecated way, "
-                    "overridden method does not accept response_body argument.",
-                    category=ScrapyDeprecationWarning,
-                )
-
-        if self._deprecated_convert_image:
-            image, buf = self.convert_image(orig_image)
-        else:
-            image, buf = self.convert_image(
-                orig_image, response_body=BytesIO(response.body)
-            )
+        image, buf = self.convert_image(
+            orig_image, response_body=BytesIO(response.body)
+        )
         yield path, image, buf
 
         for thumb_id, size in self.thumbs.items():
             thumb_path = self.thumb_path(
                 request, thumb_id, response=response, info=info, item=item
             )
-            if self._deprecated_convert_image:
-                thumb_image, thumb_buf = self.convert_image(image, size)
-            else:
-                thumb_image, thumb_buf = self.convert_image(image, size, buf)
+            thumb_image, thumb_buf = self.convert_image(image, size, response_body=buf)
             yield thumb_path, thumb_image, thumb_buf
 
     def convert_image(
         self,
         image: Image.Image,
         size: tuple[int, int] | None = None,
-        response_body: BytesIO | None = None,
+        *,
+        response_body: BytesIO,
```

Comment:
I get that the goal is to make the parameter required, without a default value, but we deprecated it having `None` as value, not it being used as a positional argument, so this change is backward-incompatible.

I’m thinking we should raise `ValueError` if the value is `None` now, and log a new warning if the parameter is used positionally, so that we can eventually make this change. And maybe we should take the chance to do the same with  `size`, deprecate it being positional.

I was unsure how to handle such a deprecation without removing the typing hints. Gemini suggested using `inspect`:

```python
import inspect

def my_function(param1: int, param2: str = "default_value"):
    frame_record = inspect.stack()[1]
    args, _, _, values = inspect.getargvalues(frame_record[0])
    if param2 in args:
        print("param2 was passed positionally")
    else:
        print("param2 was passed as a keyword argument")
```

Reply:
FTR this suggestion doesn't work :) https://stackoverflow.com/questions/56468082/inspect-if-an-argument-was-passed-positionally-or-via-keyword

Reply:
:disappointed: